amazon-kinesis-client/src/test/java/software/amazon/kinesis/retrieval/polling/PrefetchRecordsPublisherIntegrationTest.java

```diff
@@ -224,6 +225,30 @@ public DataFetcherResult answer(final InvocationOnMock invocationOnMock) throws
         verify(dataFetcher).restartIterator();
     }
 
+    @Test
+    public void testExpiredIteratorExceptionWithInnerRestartIteratorException() {
+        when(dataFetcher.getRecords()).thenAnswer(new Answer<DataFetcherResult>() {
+            @Override
+            public DataFetcherResult answer(final InvocationOnMock invocationOnMock) throws Throwable {
+                throw ExpiredIteratorException.builder().message("ExpiredIterator").build();
+            }
+        }).thenCallRealMethod();
+        doThrow(new IllegalStateException()).when(dataFetcher).restartIterator();
```

Comment:
Just throw an Exception or Throwable here so that we will be sure all class of exceptions are handled.